_source/_posts/2022-07-29-python-okta.md

```diff
@@ -0,0 +1,191 @@
+
+6. You also need to add your **Post Logout Redirect URIs**. In the developer dashboard, these are called **Sign-out redirect URIs**. Since we used localhost:3000, the URI will be: `http://localhost:3000/`.  
+
+## Run the project
+
+You can run the project now. Type:
+
+```bash
+python3 -m flask run --port=3000
+```
+
+And try walking through hosted authentication with Okta! If all goes well, you'll see this screen to start with: 
+
+{% img blog/python-okta/1_login.png alt:"Picture of log in page." width:"600" %}{: .center-image }
+
+When you click the **Login** button, you're then taken to a form that you can enter your email and password: 
+
+{% img blog/python-okta/2_sign_in.png alt:"Picture of sign in page where you add account credentials." width:"300" %}{: .center-image }
+
+You will know you've successfully logged in when you see a screen displaying some information about your profile:
+
+{% img blog/python-okta/3_success.png alt:"Picture of success page displaying retrieved account info." width:"600" %}{: .center-image }
+
+
+## Troubleshooting - Help, it's not working!
+
+Never fear, the troubleshooting section is here to help you figure out what's going wrong with your application. Look for the item that matches your problem and read on to learn how to fix it.
+
+### It says I'm not authorized to view localhost:5000
+
+If your Mac runs Monterey or later, it uses port 5000 for the AirPlay Receiver. If you have access to the *Sharing* folder on your computer, you may be able to shut it off and run the application at that point. If you want to try that, [follow these instructions](https://www.businessinsider.com/how-to-turn-off-airplay) to turn off the AirPlay Receiver.
+
+You should then be able to use `localhost:5000` again. 
+
+### I tried everything and I still can't view localhost:5000
+
+If you cannot access the *Sharing* folder on your Mac due to permissions, or you tried shutting off AirPlay Receiver and it still isn't letting you see `localhost:5000`, then change the project to use `localhost:3000`. You'll need to do these things for that to work: 
+
+1. In your copy of the project, navigate to *app.py*. Change:
+```bash
+app.run(host="localhost", port=5000, debug=True)
+```
+To: 
+```bash
+app.run(host="localhost", port=3000, debug=True)
+```
+
+2. In **app.py**, go to where it says **# get request params** and in the json for **query_params** change the **redirect_uri** from `http://localhost:5000/authorization-code/callback` to `http://localhost:3000/authorization-code/callback`.
+
+3. Log in as an administrator to your Okta developer account. Your log in URL for this will be your Organization URL, with a tweak to show you're logging in as the administrator. The URL takes the form of `https://dev-assignednumber-admin.okta.com`. 
```

Comment:
```suggestion
3. Log in as an administrator to your Okta developer account. Your login URL for this will be your Organization URL, with a tweak to show you're logging in as the administrator. The URL takes the form of `https://dev-assignednumber-admin.okta.com`. 
```
login here is adjective/one word